syntax = "proto2";

package pb;

message E4Config{
	required uint32 N = 1;
	optional double delta_0 = 2;
}

message E4State{
	required double E = 1;
	required double phi = 2;
	repeated group Particles = 3{
		required double a = 4;
		required double psi = 5;
		required double z = 6;
		required double delta = 7;
	}
}

message E4PetscSolverConfig{
	optional double atol = 1;
	optional double rtol = 2;
	optional double init_step = 3;
	optional int32 n_cores = 4;
}

message E4Model{
	required E4Config pconfig = 1;
	required E4State state = 2;
	required E4PetscSolverConfig sconfig = 3;
}

message E4Solution{
	required E4State state = 1;
	optional E4State d_state = 2;
}

message E4Special{
	optional double e_2 = 1;
	optional double aver_a_2 = 2;
	optional double int_e_a = 4;
	optional double Na = 5;
	optional double Nb = 6;
	optional double M = 17;
	optional double N = 18;
	optional double Ia = 7;
	optional double Ib = 8;
	optional double Ia_aver = 9;
	optional double Ib_aver = 10;	
	optional double Wa = 11;
	optional double Wb = 12;
}